feat: add schematics for automatic installation [AT2019-1023]

tomkoreny opened the pull request:
To test, feel free to run `npm run build && schematics .:ng-add --debug=false` in the plugin folder.

tomkoreny committed:
feat: add schematics for automatic installation [AT2019-1023]

Flyrell commented on packages/eslint-plugin-angular-template-spacing/src/collection.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
```
I'm not sure if we can rely on that path. This will work in our monorepo, but not outside of it.

Flyrell replied:
![Screen Shot 2022-09-15 at 12 55 15](https://user-images.githubusercontent.com/19550608/190386515-b70d5b78-d638-4b3d-9506-d26c1cb3d81d.png)
That's what ngrx has in their repo. Maybe we can somehow reuse that? But I don't know about the difference between `@angular-devkit/schematics` and `@schematics/angular`.

tomkoreny replied:
I don't think we want to extend angulars schematics. However the `$schema` key is just for schema validation, i pointed it directly to git url.

Flyrell commented on packages/eslint-plugin-angular-template-spacing/src/schematics.ts:
```diff
@@ -0,0 +1,46 @@
+    const path = '.eslint.json';
+    const pluginName = '@betsys-eslint/eslint-plugin-angular-template-spacing'
+    const pluginSettings = `plugin:${pluginName}/recommended`
+    const pluginUrl = 'https://github.com/betsys-com/betsys-eslint/tree/main/packages/eslint-plugin-angular-template-spacing'
```
Can't we read that info like name/url from package.json? This would be more reliable.

Flyrell commented on packages/eslint-plugin-angular-template-spacing/src/schematics.ts:
```diff
@@ -0,0 +1,46 @@
+      return tree;
+    }
+
+    createOrPushToArray(json, 'plugins', pluginName)
```
You can have your `.esling.json` file that looks like this:
```json5
{
  // ...
  overrides: [
    {
      files: ['*.ts'],
      plugins: [/* ... */],
      extends: [/* ... */],
    }
  ]
}
```

if the specific override exists for the `*.ts` files, we should add it there instead of root.

Flyrell commented on packages/eslint-plugin-angular-template-spacing/src/schematics.ts:
```diff
@@ -0,0 +1,46 @@
+    createOrPushToArray(json, 'plugins', pluginName)
+    createOrPushToArray(json, 'extends', pluginSettings)
+
+    tree.overwrite(path, JSON.stringify(json, null, "\t"))
```
We use 4 spaces usually, instead of tabs, would that work?

tomkoreny replied:
I don't know if there is any way to see what the preference of the repository is. We can use whatever we choose here, just need to agree on something in the old tabs vs spaces debate.

Flyrell commented on packages/eslint-plugin-angular-template-spacing/rollup.config.js:
```diff
@@ -1,21 +1,34 @@
 import json from '@rollup/plugin-json';
 import { createBaseRollupConfig } from '../../rollup.helper';
+import commonjs from "@rollup/plugin-commonjs";
```
what is this plugin for?

tomkoreny replied:
Without it, i keep getting this [error](https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module) when bundling the schematics. Any other way to solve it?

tomkoreny committed:
fix: use override when applicable and point json schema directly to url

Flyrell committed:
feat(angular-template-spacing): cosmetic changes

Flyrell committed:
feat(install): schematics

Flyrell committed:
feat(angular-template-spacing): schematics

Flyrell committed:
Publish

 - @betsys-eslint/eslint-plugin-angular-template-spacing@1.0.2-alpha.0
 - @betsys-eslint/install@0.1.1-alpha.0

Flyrell committed:
test(install): schema installer

Flyrell committed:
feat(angular-template-spacing): cosmetic changes

Flyrell committed:
Publish

 - @betsys-eslint/eslint-plugin-angular-template-spacing@1.0.2
 - @betsys-eslint/install@0.2.0

Flyrell committed:
fix(angular-template-spacing): remove unused dep
